streaming/streaming.go

```diff
@@ -0,0 +1,258 @@
+			return fmt.Errorf("error generating stream name: %v", err)
+		}
+		streamingCfg.DataflowCfg.JobName = jobName
+	}
+	return nil
+}
+
+// ReadStreamingConfig reads the file and unmarshalls it into the StreamingCfg struct.
+func ReadStreamingConfig(file, dbName string) (StreamingCfg, error) {
+	streamingCfg := StreamingCfg{}
+	cfgFile, err := ioutil.ReadFile(file)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("can't read streaming config file due to: %v", err)
+	}
+	err = json.Unmarshal(cfgFile, &streamingCfg)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("unable to unmarshall json due to: %v", err)
+	}
+	err = VerifyAndUpdateCfg(&streamingCfg, dbName)
+	if err != nil {
+		return streamingCfg, fmt.Errorf("streaming config is incomplete: %v", err)
+	}
+	return streamingCfg, nil
+}
+
+// LaunchStream populates the parameters from the streaming config and triggers a stream on Cloud Datastream.
+func LaunchStream(ctx context.Context, dbName, projectID string, datastreamCfg DatastreamCfg) error {
+	fmt.Println("Launching stream...")
+	dsClient, err := datastream.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("datastream client can not be created: %v", err)
+	}
+	defer dsClient.Close()
+	fmt.Println("Created client...")
+
+	mydb := &datastreampb.MysqlDatabase{
+		DatabaseName: dbName,
+	}
+	mysqlSrcCfg := &datastreampb.MysqlSourceConfig{
+		Allowlist: &datastreampb.MysqlRdbms{MysqlDatabases: []*datastreampb.MysqlDatabase{mydb}},
+	}
+	gcsDstCfg := &datastreampb.GcsDestinationConfig{
+		Path:       datastreamCfg.DestinationConnectionConfig.Prefix,
+		FileFormat: &datastreampb.GcsDestinationConfig_AvroFileFormat{},
+	}
+	srcCfg := &datastreampb.SourceConfig{
+		SourceConnectionProfileName: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectID, datastreamCfg.SourceConnectionConfig.Location, datastreamCfg.SourceConnectionConfig.Name),
+		SourceStreamConfig:          &datastreampb.SourceConfig_MysqlSourceConfig{MysqlSourceConfig: mysqlSrcCfg},
+	}
+	dstCfg := &datastreampb.DestinationConfig{
+		DestinationConnectionProfileName: fmt.Sprintf("projects/%s/locations/%s/connectionProfiles/%s", projectID, datastreamCfg.DestinationConnectionConfig.Location, datastreamCfg.DestinationConnectionConfig.Name),
+		DestinationStreamConfig:          &datastreampb.DestinationConfig_GcsDestinationConfig{GcsDestinationConfig: gcsDstCfg},
+	}
+	streamInfo := &datastreampb.Stream{
+		DisplayName:       datastreamCfg.StreamDisplayName,
+		SourceConfig:      srcCfg,
+		DestinationConfig: dstCfg,
+		State:             datastreampb.Stream_RUNNING,
+		BackfillStrategy:  &datastreampb.Stream_BackfillNone{BackfillNone: &datastreampb.Stream_BackfillNoneStrategy{}},
+	}
+	createStreamRequest := &datastreampb.CreateStreamRequest{
+		Parent:   fmt.Sprintf("projects/%s/locations/%s", projectID, datastreamCfg.StreamLocation),
+		StreamId: datastreamCfg.StreamId,
+		Stream:   streamInfo,
+	}
+
+	fmt.Println("Created stream request..")
```

Comment:
add details of stream request

Reply:
As mentioned above, we print out the whole input body if things go wrong.